storage/innobase/row/row0ins.cc

```diff
@@ -429,7 +429,8 @@ row_ins_cascade_ancestor_updates_table(
 
 		upd_node = static_cast<upd_node_t*>(parent);
 
-		if (upd_node->table == table && upd_node->is_delete == FALSE) {
+		if (upd_node->table == table && upd_node->is_delete == FALSE
+			&& !upd_node->vers_delete) {
```

Comment:
Please replace the `== FALSE` suffix with `!` prefix, and indent the 2nd line with 1 less TAB and 4 more spaces, so that it aligns with the previous line.

Please add a source code comment to explain why the condition on `upd_node->vers_delete` is needed. I see that the tests in versioning.cte could exercise this code.

Can we have a member function, say, `upd_node->deleting()` that checks both flags, and also contains a debug assertion that at most  one of them may be set?

Maybe the `is_delete` should be a 3-valued `enum` instead of `bool`?

Reply:
Related to https://github.com/tempesta-tech/mariadb/issues/101. Foreign constrainsts should be checked only against current rows, so `vers_delete` restricts it to this case. Agree that `enum` is better.